tests/test_sqlcompletion.py

```diff
@@ -99,14 +99,42 @@ def test_suggests_tables_views_and_schemas(expression):
 
 @pytest.mark.parametrize('expression', [
     'SELECT * FROM ',
-    'SELECT * FROM foo JOIN ',
 ])
-def test_suggest_tables_views_schemas_and_set_returning_functions(expression):
+def test_suggest_tables_views_schemas_and_functions(expression):
     suggestions = suggest_type(expression, expression)
     assert set(suggestions) == set([
         Table(schema=None),
         View(schema=None),
         Function(schema=None, filter='for_from_clause'),
+        Schema()
+    ])
+
+
+@pytest.mark.parametrize('expression', [
+    'SELECT * FROM foo JOIN bar on bar.barid = foo.barid JOIN ',
+    'SELECT * FROM foo JOIN bar USING (barid) JOIN ',
+])
+def test_suggest_after_join_with_two_tables(expression):
+    suggestions = suggest_type(expression, expression)
+    assert set(suggestions) == set([
+        Table(schema=None),
+        View(schema=None),
+        Function(schema=None, filter='for_from_clause'),
+        Join(((None, 'foo', None, False), (None, 'bar', None, False)), None),
+        Schema(),
+    ])
+
+
+@pytest.mark.parametrize('expression', [
+    'SELECT * FROM foo JOIN '
```

Comment:
Can you add one extra condition here like `'SELECT * FROM foo JOIN bar'`. That passes with your current PR, but fails if we revert the changes to the `text_before_cursor` stuff.